florist/api/routes/server/job.py

```diff
@@ -45,3 +45,29 @@ async def new_job(request: Request, job: Job = Body(...)) -> Dict[str, Any]:  #
     assert isinstance(created_job, dict)
 
     return created_job
+
+
+@router.get(path="/{status}", response_description="List jobs with the specified status", response_model=List[Job])
+async def list_jobs_with_status(status: JobStatus, request: Request) -> List[Dict[str, Any]]:
+    """
+    List jobs with specified status.
+
+    Fetches list of Job with max length MAX_RECORDS_TO_FETCH.
+
+    :param status: (JobStatus) The status of jobs to query the Job DB for.
+    :param request: (fastapi.Request) the FastAPI request object.
+    :return: (List[Dict[str, Any]]) A list where each entry is a dictionary with the attributes
+        of a Job instance with the specified status.
+    :raises: (HTTPException) status 400 if status is not a member of the JobStatus Enum.
+    """
+    status = jsonable_encoder(status)
+
+    assert isinstance(status, str)
+    if status not in JobStatus._member_names_:
```

Comment:
It's better not to use properties with underscore in the beginning. For the Client enum and others, I made a `list` function to save a few lines of code, the same can be done here as well.

https://github.com/VectorInstitute/FLorist/blob/main/florist/api/clients/common.py#L30-L36

Reply:
Kinda crazy python doesn't have something like this to save us from doing list comprehensions everywhere.